src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](/convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const startConvertion = sourceString
+    .split(';')
+    .map(property => property.trim());
+
+  const removeEmptyStrings = startConvertion.filter(property =>
+    (property.length > 0));
+
+  const formattedStyle = removeEmptyStrings.reduce((acc, currentProperty) => {
```

Comment:
`reduce` allows to put all logic inside itself, try to `split` entry value by `;` and after that use `reduce` with whole logic inside